File: 04.Simulation模拟/leetcode_2161_pivotArray.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> res;
        int indx1 = 0, indx2 = 0;
        for (int i : nums){
            if (i < pivot){
                res.insert(res.begin()+indx1, i);
                indx1++;
                indx2++;
            }
            else if (i == pivot){
                res.insert(res.begin()+indx2, i);
                indx2++;
            }
            else{
                res.push_back(i);
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {9,12,5,10,14,3,10};
    int pivot = 10;
    vector<int> res = s.pivotArray(nums, pivot);
    for (int i : res){
        cout << i << " ";
    }
    cout << endl;
}

File: 04.Simulation模拟/leetcode_2161_pivotArray.py
from typing import List

class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        res1, res2, res3 = [], [], []
        for i in nums:
            if i < pivot:
                res1.append(i)
            elif i == pivot:
                res2.append(i)
            else:
                res3.append(i)
        return res1+res2+res3

if __name__ == "__main__":
    s = Solution()
    nums = [9,12,5,10,14,3,10]
    pivot = 10
    print(s.pivotArray(nums, pivot))
